docs/advanced-features/athena.ipynb

```diff
@@ -0,0 +1,688 @@
+    "      regexp_extract(\"$path\", '{S3_MATCH}', 4) as user,\n",
+    "      regexp_extract(\"$path\", '{S3_MATCH}{SLASH}', 5) as name,\n",
+    "      regexp_extract(\"$path\", '[^/]+$') as timestamp,\n",
+    "      {PACKAGES_TABLE}.\"hash\"\n",
+    "      FROM {PACKAGES_TABLE}\n",
+    "  ),\n",
+    "  mv AS (\n",
+    "    SELECT\n",
+    "      regexp_extract(\"$path\", '[^/]+$') as tophash,\n",
+    "        manifest.\"meta\",\n",
+    "        manifest.\"message\"\n",
+    "      FROM\n",
+    "        {MANIFEST_TABLE} as manifest\n",
+    "      WHERE manifest.\"logical_key\" IS NULL\n",
+    "  )\n",
+    "SELECT\n",
+    "  npv.\"user\",\n",
+    "  npv.\"name\",\n",
+    "  npv.\"hash\",\n",
+    "  npv.\"timestamp\",\n",
+    "  mv.\"message\",\n",
+    "  mv.\"meta\"\n",
+    "FROM npv\n",
+    "JOIN\n",
+    "  mv\n",
+    "ON\n",
+    "  npv.\"hash\" = mv.\"tophash\"\n",
+    "\"\"\"\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2111bfa0-ccea-4169-aeda-b67b8b317ed9",
+   "metadata": {},
+   "source": [
+    "### D. View of object-Level metadata\n",
+    "The DDL below creates a view that contains package contents, including:\n",
+    "* logical_key\n",
+    "* physical_keys\n",
+    "* object hash\n",
+    "* object metadata\n",
+    "<!--pytest-codeblocks:cont-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "id": "04648076-121a-488a-abd3-c8c0345e92a9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "DDL[OBJECTS_VIEW] = f\"\"\"\n",
+    "CREATE OR REPLACE VIEW {ATHENA_DB}.{OBJECTS_VIEW} AS\n",
+    "WITH\n",
+    "  mv AS (\n",
+    "    SELECT\n",
+    "      regexp_extract(\"$path\", '[^/]+$') as tophash,\n",
+    "      manifest.\"logical_key\",\n",
+    "      manifest.\"physical_keys\",\n",
+    "      manifest.\"size\",\n",
+    "      manifest.\"hash\",\n",
+    "      manifest.\"meta\",\n",
+    "      manifest.\"user_meta\"\n",
+    "    FROM\n",
+    "      {MANIFEST_TABLE} as manifest\n",
+    "    WHERE manifest.\"logical_key\" IS NOT NULL\n",
+    "  )\n",
+    "SELECT\n",
+    "  npv.\"user\",\n",
+    "  npv.\"name\",\n",
+    "  npv.\"timestamp\",\n",
+    "  mv.\"tophash\",\n",
+    "  mv.\"logical_key\",\n",
+    "  mv.\"physical_keys\",\n",
+    "  mv.\"hash\",\n",
+    "  mv.\"meta\",\n",
+    "  mv.\"user_meta\"\n",
+    "FROM mv\n",
+    "JOIN\n",
+    "  {PACKAGES_VIEW} as npv\n",
+    "ON\n",
+    "  npv.\"hash\" = mv.\"tophash\"\n",
+    "\"\"\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "87935de8-2342-4712-b8da-e60e34d56d87",
+   "metadata": {},
+   "source": [
+    "You can run the following Python code to create the preceding tables and views:\n",
+    "<!--pytest-codeblocks:cont-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "id": "1ee63b7a-ebf0-4ab0-9a79-899fc0c2238e",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\n",
```

Comment:
Why do we need this newline?

Reply:
Because when the code is being tested via purest, it coalesces all the code blocks together, and makes it hard to see which section is doing what. The initial newline makes it clearer visually:

> print("\nTest Athena Query:")